lib/adapters/http_server.js

```diff
@@ -0,0 +1,133 @@
+var http = require('http');
+var _ = require('underscore');
+var contentType = require('content-type');
+
+var Juttle = require('../runtime/index').Juttle;
+var parsers = require('./parsers');
+
+var Read = Juttle.proc.source.extend({
+    procName: 'read-http_server',
+
+    initialize: function(options, params, location, program) {
+        var allowed_methods = [ 'POST', 'PUT' ];
+
+        var allowed_options = [
+            'port',
+            'method',
+            'timeField'
+        ];
+
+        var unknown = _.difference(_.keys(options), allowed_options);
+        if (unknown.length > 0) {
+            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
+                proc: 'read http_server',
+                option: unknown[0]
+            });
+        }
+
+        if (options.method && allowed_methods.indexOf(options.method) === -1) {
+            throw this.compile_error('RT-INVALID-OPTION-VALUE', {
+                option: 'method',
+                supported: allowed_methods.join(', ')
+            });
+        }
+
+        this.port = options.port || 8080;
+        this.method = options.method || 'POST';
+        this.timeField = options.timeField;
+
+        if (params.filter_ast) {
+            throw this.compile_error('RT-ADAPTER-UNSUPPORTED-FILTER', {
+                proc: 'read http_server',
+                filter: 'filtering'
+            });
+        }
+    },
+
+    start: function() {
+        var self = this;
+        var format;
```

Comment:
With `format` scoped here, it will be preserved across requests, so if you make a request with one content type and then the next request's content type throws an error on line 56 `format` will be the old content type instead of `undefined`.